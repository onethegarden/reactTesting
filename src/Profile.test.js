import React from "react";
import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile/>", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile username="jeongwonHan" name="한정원" />);
    expect(wrapper).toMatchSnapshot();
  });
  it("renders username and name", () => {
    const wrapper = mount(<Profile username="jeongwonHan" name="한정원" />);
    expect(wrapper.props().username).toBe("jeongwonHan");
    expect(wrapper.props().name).toBe("한정원");

    const boldElement = wrapper.find("b");
    expect(boldElement.contains("jeongwonHan")).toBe(true);
    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("한정원");
  });
});
