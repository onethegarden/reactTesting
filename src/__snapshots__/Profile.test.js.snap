// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> matches snapshot 1`] = `
<Profile
  name="한정원"
  username="jeongwonHan"
>
  <div>
    <b>
      jeongwonHan
    </b>
     
    <span>
      한정원
    </span>
  </div>
</Profile>
`;
